### PR TITLE
Kmeans cuda

### PR DESCRIPTION
**Pull request by liamlin5566**
- add cuda code (src/kernelfunc.cu)
- modify makefile to be compiled with cuda (result can be seen in image1_result_cuda.png)
- add .gitignore, remove namespace

### CODE REVIEW

**Comment by liamlin5566** [comment]
Hi @yungyuc, I modify some and add cuda implementation. Could you take some time to review if you are available?  Thanks.

File: proj/src/kmeans.cpp [hunk 1]
```diff
@@ -289,3 +297,128 @@ std::vector<int> argmin_omp(std::vector<std::vector<double>> &vec, int nthreads)
     return min_indices;
 }
 
+
+
+#ifdef WCUDA
```
Comment (yungyuc) [inline proj/src/kmeans.cpp:302]: You probably may want to know https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#nvcc-identification-macro

File: proj/pyKmeans.cpp [hunk 2]
```diff
@@ -84,11 +98,14 @@ class pyKmeans{
...
 };
 
+}/*end namespace pybind11*/
+
 
 PYBIND11_MODULE(_kmeans, m) 
 {
     m.doc() = "K-means on image";
 
+    using namespace pybind11;
```
Comment (yungyuc) [inline proj/pyKmeans.cpp:108]: If you `using namespace pybind11;`, why bother writing `py::...`?  The inconsistency in coding style makes your code harder to understand.

**Review by yungyuc — COMMENTED** [review]
To get more effective review, you should add information with comments (annotations).

**Comment by liamlin5566** [comment]
@yungyuc . OK, I will modify the code. Thank you.

**Pull request merged**